helix-core/src/main/java/org/apache/helix/tools/ClusterVerifiers/BestPossibleExternalViewVerifier.java

```diff
@@ -274,8 +276,22 @@ protected synchronized boolean verifyState() {
 
       // Filter resources if requested
       if (_resources != null && !_resources.isEmpty()) {
-        idealStates.keySet().retainAll(_resources);
-        extViews.keySet().retainAll(_resources);
+        // Find waged-enabled resources among the requested resources
+        Set<String> requestedWagedResources = _resources.stream().filter(
```

Comment:
If we only want to know wether there are waged resources or not, we dont need to iterate through all resources. We could just iterate and return a bool. (Also we can do early exit if we find one.)